controllers/apps/transformer_component_rbac.go

```diff
@@ -197,15 +174,11 @@ func buildServiceAccount(transCtx *componentTransformContext) (*corev1.ServiceAc
 		compDef         = transCtx.CompDef
 		synthesizedComp = transCtx.SynthesizeComponent
 	)
-
-	// TODO(component): dependency on cluster definition
-	backupPolicyTPL, err := getDefaultBackupPolicyTemplate(transCtx, cluster.Spec.ClusterDef)
+	serviceAccountName := comp.Spec.ServiceAccountName
+	dataProtectionEnable, err := isDataProtectionEnabled(transCtx, compDef)
 	if err != nil {
 		return nil, err
 	}
-
-	serviceAccountName := comp.Spec.ServiceAccountName
-	dataProtectionEnable := isDataProtectionEnabled(backupPolicyTPL, cluster, comp)
 	if serviceAccountName == "" {
 		// If lifecycle actions and data protection are disabled at the same tme, then do not create a service account.
```

Comment:
@wangyelei @ldming  Do you know why a service account should be created for dataProtection?

Reply:
need a backup update role.

Reply:
The role should be configured through the policyRules of cmpd.